### PR TITLE
Features/entities/api get versions

### LINKED ISSUE #799: Get version history of an entity
`GET /v1/projects/:projectId/datasets/:name/entities/:uuid/versions`
Extended: expand creator into actor, include data

Should follow
https://github.com/getodk/central-backend/issues/798 
https://github.com/getodk/central-backend/issues/797

### PR DESCRIPTION
**Pull request by sadiqkhoja**
Closes #799

#### What has been done to verify that this works as intended?

Integration Tests

#### Why is this the best possible solution? Were any other approaches considered?

Lot of discussions, we may need to revisit source information once we work on PUT/POST endpoints

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?

No regression risks.

#### Does this change require updates to the API documentation? If so, please update docs/api.md as part of this PR.

There is an open PR, will update it once implementation of entity source is completed.

#### Before submitting this PR, please make sure you have:

- [x] run `make test-full` and confirmed all checks still pass OR confirm CircleCI build passes
- [x] verified that any code from external sources are properly credited in comments or that everything is internally sourced

### CODE REVIEW

File: lib/model/query/entities.js [hunk 1]
```diff
@@ -173,6 +173,47 @@ const getAll = (datasetId, options = QueryOptions.none) => ({ all }) =>
   _get(false)(all, options.withCondition({ datasetId }), isTrue(options.argData.deleted))
     .then(map((e) => e.withAux('currentVersion', e.aux.currentVersion.withAux('creator', e.aux.currentVersionCreator))));
 
+const getByUuid = (uuid) => ({ maybeOne }) =>
+  maybeOne(sql`select * from entities where "uuid" = ${uuid} limit 1`)
+    .then(map(construct(Entity)));
+
```
Comment (sadiqkhoja) [inline lib/model/query/entities.js:179]: this can be deleted.

File: lib/model/query/entities.js [hunk 2]
```diff
@@ -173,6 +173,47 @@ const getAll = (datasetId, options = QueryOptions.none) => ({ all }) =>
...
+  SELECT ${fields} FROM entities
+  JOIN entity_defs ON entities.id = entity_defs."entityId"
+  LEFT JOIN submission_defs ON submission_defs.id = entity_defs."submissionDefId"
+  LEFT JOIN (
+    SELECT submissions.*, submission_defs."userAgent" FROM submissions
+    JOIN submission_defs ON submissions.id = submission_defs."submissionId" AND root
+  ) submissions ON submissions.id = submission_defs."submissionId"
+  LEFT JOIN forms ON submissions."formId" = forms.id
+  ${extend||sql`
+    LEFT JOIN actors ON actors.id=entities."creatorId"
```
Comment (sadiqkhoja) [inline lib/model/query/entities.js:193]: this join is not required.

File: lib/model/query/entities.js [hunk 3]
```diff
@@ -173,6 +173,47 @@ const getAll = (datasetId, options = QueryOptions.none) => ({ all }) =>
...
+    LEFT JOIN actors current_version_actors ON current_version_actors.id=entity_defs."creatorId"
```
Comment (sadiqkhoja) [inline lib/model/query/entities.js:194]: change the alias.

File: test/integration/api/entities.js [hunk 4]
```diff
@@ -201,10 +201,36 @@ describe('Entities API', () => {
...
+      await asChelsea.get('/v1/projects/1/datasets/people/entities/12345678-1234-4123-8234-123456789abc')
+        .expect(403);
+    }));
+
+    it('should return all versions of the Entity', testEntities(async (service, container) => {
       const asAlice = await service.login('alice');
 
-      await asAlice.get('/v1/projects/1/datasets/People/entities/00000000-0000-0000-0000-000000000001/versions')
+      await container.db.any(sql`UPDATE entity_defs SET current = false;`);
+      await container.db.any(sql`
```
Comment (sadiqkhoja) [inline test/integration/api/entities.js:229]: add a comment to change this temp query to update API

File: test/integration/api/entities.js [hunk 5]
```diff
@@ -201,10 +201,36 @@ describe('Entities API', () => {
...
+        INSERT INTO entity_defs ("entityId", "createdAt", "current", "submissionDefId", "data", "creatorId", "userAgent", "label")
+        SELECT "entityId", clock_timestamp(), TRUE, NULL, "data", "creatorId", 'postman', "label" || ' updated' FROM entity_defs`);
```
Comment (sadiqkhoja) [inline test/integration/api/entities.js:231]: use different creator and assert.

**Review by ktuite — APPROVED** [review]
Just did interactive review with @sadiqkhoja

**Pull request merged**